### DISCUSSION TITLE
Re: Newtonsoft what was the reasoning for Legacy

### DISCUSSION THREAD

**Discussion by mwpowellhtx** [OrchardCMS/OrchardCore.Commerce#104] [Q&A] [answered]
I see there were some efforts, apparently, to transition to core `System` based `Json` serialization? `Newtonsoft` was not getting the job done, or what was the story behind that, please...

Perhaps also elaborate a bit on the comments concerning serialization of incomplete amounts.

Thank you.

**Comment by bleroy** [comment, marked as answer]
System can be much faster and allocate less memory, so I wanted to try it out, but the timing wasn't right, and Orchard Core itself has not made that transition. In the end there wasn't much point in doing it, so I reverted it.

**Reply by mwpowellhtx** [reply]
Fair enough, thank you for the insights.

**Reply by mwpowellhtx** [reply]
Finding however in the _Commerce_ assembly itself, apparently rolled forward with the _Microsoft_ `System` _Json serialization_? Which again fair enough; it was _'ready'_ enough for prime time, and the _Newtonsoft_ was there as a _legacy contingency_?

**Reply by bleroy** [reply]
Not sure what you're asking.

**Reply by mwpowellhtx** [reply]
Same question. Wondering where _OC CMS Commerce_ fits in the _JSON serialization story roadmap_. It appears `System` is predominantly in use, especially taking into consideration the parent assembly. Is this accurate?

**Reply by bleroy** [reply]
I'm sorry, I'm not following. What are you getting at? What JSON serialization roadmap? What do you mean by "parent assembly"?

**Reply by mwpowellhtx** [reply]
@bleroy I'm sorry I thought you had more oversight on the project, repo, etc, than this.

* _Newtonsoft_ was/is used for the _DataTypes_ projects; and at least designated as a _"Legacy"_ concern.
* `OrchardCore.Commerce` is the primary _OC CMS Module_ assembly of concern in the project.
  * In which I am finding _serialization_ throughout that written exclusively for `System` _JSON_ support; have I misread that?

So, in terms of puzzle pieces, just trying to gain a better understanding where the _Newtonsoft/System_ question fits in that timeline.

I have not looked more closely at the commit history, however, at this time.

Thank you...

**Reply by bleroy** [reply]
I seem to have misremembered it. I thought I had rolled back most of the `System.Text.JSON` changes because it didn't seem worth the trouble maintaining two serialization mechanisms (since Newtonsoft needed to be kept around to properly interop with the main Orchard Core project at the time - that was about 2 years ago - ). Seems like I never did and that Newtonsoft is still labeled as "legacy".

**Reply by mwpowellhtx** [reply]
Been there myself over my career; I understand. So, obviously, _'safe'_ to assume, as it were, `System` is the path forward, at least as far as today's revision of _OC CMS Commerce_ is concerned.

**Reply by bleroy** [reply]
Let me check with some Orchard Core maintainers what the status is on their side and I'll get back to you. We should be consistent with that.

**Reply by Skrypt** [reply]
Unless I'm wrong, I don't think System.Text.Json has everything yet that we use in Newtonsoft. Maybe it changed since last time because I don't remember exactly what right now. I'm pretty sure @sebastienros and @deanmarcussen will remember.

**Reply by mwpowellhtx** [reply]
Thanks so much @Skrypt.

For my own edification, I've loosely followed the _System/Newtonsoft JSON_ epic OSS meetings, and that jives somewhat with my high level comprehension. Now, what that means for _OC_, _CMS_, _Commerce_, is an entirely different discussion; i.e. whether those Newtonsoft features are, in fact, leveraged. I cannot say.

**Reply by bleroy** [reply]
So currently, Orchard Core still relies mostly on Newtonsoft, but the goal eventually is to use System.Text.Json. That means that it makes sense for Core Commerce to have both types of converters maintained in the codebase to be ready for that transition, whenever it happens even if that means a little additional maintenance overhead.

**Reply by mwpowellhtx** [reply]
@bleroy Well, that's the clarification, I suppose. There is definitely a limited dependency there, and a _'legacy'_ usage (_quot unquote_), but I think if you examine a little further, not all the serialization has that, and is in fact using the `System.Text.Json`. Which is fine, either way, I was just curious more than anything else. If there was an advantage one way or another, might be interesting to learn what that was/is moving forward. Anyway, thanks for looking into it a bit.

**Reply by bleroy** [reply]
Hahahaha. Indeed, some objects have only System.Text.Json support. Objects that don't interact with the underlying Orchard Core infrastructure, and as such don't have the same requirements.

**Reply by mwpowellhtx** [reply]
For scope I counted perhaps _two classes_ which had _'legacy'_ alternatives. Any of the others were `System`. 🍻 Cheers.

**Reply by bleroy** [reply]
Yes, amounts and addresses need to be persisted as properties of content parts, for which persistence is handled by Orchard Core, so Newtonsoft support is necessary. For types that are not used as content part or field properties, we handle the serialization 100% and as such only need System.
